### ISSUE TITLE
make error

### ISSUE THREAD

**Issue by p1tach10** [protobuf-c/protobuf-c#275]
after I installed all required packet and finished run ./autogen.sh && ./configure 
when I run make file there was an error.

```sh
/src/protobuf-c$ make
  CXX      protoc-c/protoc_c_protoc_c-c_helpers.o
protoc-c/c_helpers.cc: In function ‘void google::protobuf::compiler::c::SplitStringUsing(const string&, const char*, std::vector<std::__cxx11::basic_string<char> >*)’:
protoc-c/c_helpers.cc:507:3: error: ‘back_insert_iterator’ was not declared in this scope
   back_insert_iterator< vector<string> > it(*result);
   ^
protoc-c/c_helpers.cc:507:3: note: suggested alternative:
In file included from /usr/include/c++/5/bits/stl_algobase.h:67:0,
                 from /usr/include/c++/5/vector:60,
                 from protoc-c/c_helpers.cc:63:
/usr/include/c++/5/bits/stl_iterator.h:415:11: note:   ‘std::back_insert_iterator’
     class back_insert_iterator
           ^
protoc-c/c_helpers.cc:507:40: error: expected primary-expression before ‘>’ token
   back_insert_iterator< vector<string> > it(*result);
                                        ^
protoc-c/c_helpers.cc:507:52: error: ‘it’ was not declared in this scope
   back_insert_iterator< vector<string> > it(*result);
                                                    ^
Makefile:1271: recipe for target 'protoc-c/protoc_c_protoc_c-c_helpers.o' failed
make: *** [protoc-c/protoc_c_protoc_c-c_helpers.o] Error 1

```

Tested on:
Ubuntu 16.04

**Comment by jiangivan** [comment]
I met you the same problem, you have solved it?

**Comment by canty37** [comment]
I can get the build to finish on ubuntu 16.04 by:
1.  change back_insert_iterator on line 507 in protoc-c/c_helpers.cc to std::back_insert_iterator
2.  when you get the next build error, edit t/test-full.pb.cc and change all "static_cast<::google" to "static_cast< ::google".  There needs to be a space after the "<".

**Comment by jiangivan** [comment]
thank you very much!

**Comment by canty37** [comment]
Glad I could help!

**Comment by edmonds** [comment]
Should be fixed by #280.
